Yes, that's what I did, but this was the result:

```
 1) GET /challenges/:challengeId public guild should return challenge data:

      AssertionError: expected { Object (_id, leader, ...) } to deeply equal { Object (_id, id, ...) }
      + expected - actual

       {
         "_id": "f5f53ac8-345b-45cf-84c4-9ee5fcb112b9"
         "id": "f5f53ac8-345b-45cf-84c4-9ee5fcb112b9"
      -  "leader": "3c7c1ee9-2acc-4d51-b03b-1d4bfbda1739"
      +  "leader": {
      +    "_id": "3c7c1ee9-2acc-4d51-b03b-1d4bfbda1739"
      +    "profile": {
      +      "name": "cc022041-01a6-423e-aede-e65922fd46e6"
      +    }
      +  }
         "name": "test group"
         "privacy": "public"
         "type": "guild"
       }
```

--- a/test/api/v3/integration/challenges/GET-challenges_challengeId.test.js
+++ b/test/api/v3/integration/challenges/GET-challenges_challengeId.test.js
@@ -46,7 +46,14 @@ describe('GET /challenges/:challengeId', () => {
         id: groupLeader._id,
         profile: {name: groupLeader.profile.name},
       });
-      expect(chal.group).to.eql(_.pick(group, ['_id', 'id', 'name', 'type', 'privacy']));
+      expect(chal.group).to.eql({
+        _id: group._id,
+        id: group.id,
+        name: group.name,
+        type: group.type,
+        privacy: group.privacy,
+        leader: groupLeader.id,
+      });
     });
   });
 
@@ -91,7 +98,14 @@ describe('GET /challenges/:challengeId', () => {
         id: groupLeader._id,
         profile: {name: groupLeader.profile.name},
       });
-      expect(chal.group).to.eql(_.pick(group, ['_id', 'id', 'name', 'type', 'privacy']));
+      expect(chal.group).to.eql({
+        _id: group._id,
+        id: group.id,
+        name: group.name,
+        type: group.type,
+        privacy: group.privacy,
+        leader: groupLeader.id,
+      });
     });
   });
 
@@ -136,7 +150,14 @@ describe('GET /challenges/:challengeId', () => {
         id: groupLeader.id,
         profile: {name: groupLeader.profile.name},
       });
-      expect(chal.group).to.eql(_.pick(group, ['_id', 'id', 'name', 'type', 'privacy']));
+      expect(chal.group).to.eql({
+        _id: group._id,
+        id: group.id,
+        name: group.name,
+        type: group.type,
+        privacy: group.privacy,
+        leader: groupLeader.id,
+      });
     });
   });
 });

--- a/test/api/v3/integration/challenges/GET-challenges_user.test.js
+++ b/test/api/v3/integration/challenges/GET-challenges_user.test.js
@@ -45,6 +45,7 @@ describe('GET challenges/user', () => {
         type: publicGuild.type,
         privacy: publicGuild.privacy,
         name: publicGuild.name,
+        leader: publicGuild.leader._id,
       });
     });
 
@@ -64,6 +65,7 @@ describe('GET challenges/user', () => {
         type: publicGuild.type,
         privacy: publicGuild.privacy,
         name: publicGuild.name,
+        leader: publicGuild.leader._id,
       });
       let foundChallenge2 = _.find(challenges, { _id: challenge2._id });
       expect(foundChallenge2).to.exist;
@@ -78,6 +80,7 @@ describe('GET challenges/user', () => {
         type: publicGuild.type,
         privacy: publicGuild.privacy,
         name: publicGuild.name,
+        leader: publicGuild.leader._id,
       });
     });
 
@@ -97,6 +100,7 @@ describe('GET challenges/user', () => {
         type: publicGuild.type,
         privacy: publicGuild.privacy,
         name: publicGuild.name,
+        leader: publicGuild.leader._id,
       });
       let foundChallenge2 = _.find(challenges, { _id: challenge2._id });
       expect(foundChallenge2).to.exist;
@@ -111,6 +115,7 @@ describe('GET challenges/user', () => {
         type: publicGuild.type,
         privacy: publicGuild.privacy,
         name: publicGuild.name,
+        leader: publicGuild.leader._id,
       });
     });
 

